### PR TITLE
Jkc ps edits

### PR DESCRIPTION
**Pull request by techbunny**
Edits to Peoplesoft:  Addition of ID servers, VM extension and related changes.

### CODE REVIEW

File: Peoplesoft/modules/compute/compute.variables.tf [hunk 1]
```diff
@@ -81,4 +81,7 @@ variable "public_ip_address_allocation" {
 }
 variable "backendpool_id" {
   default = "0"
+}
+variable "create_vm" {
```
Comment (romitgirdhar) [inline Peoplesoft/modules/compute/compute.variables.tf]: I don't see this variable being used??

File: Peoplesoft/provider.tf [hunk 2]
```diff
@@ -5,9 +5,8 @@ provider "azurerm" {
     version= ">=1.21.0"
     tenant_id       = "${var.tenant_id}"
     subscription_id = "${var.subscription_id}"
-    #TODO: for testing, leaving these out ... easier to sign in via CLI.
-  #  client_id       = "${var.client_id}"
-  #  client_secret   = "${var.client_secret}"   
+    # client_id       = "${var.client_id}"
```
Comment (romitgirdhar) [inline Peoplesoft/provider.tf]: Uncomment these.

File: Peoplesoft/variables.tf [hunk 3]
```diff
@@ -5,8 +5,18 @@ variable "subscription_id" {
...
 
+variable "client_secret" {
+  
+}
+
+variable "client_id" {
+}
+variable "vnet_resource_group_name" {
+    default = "ps-rg3"
+    description = "Name of the Resource Group where the VNET is/will be located. This can be same as the app resource group or different."
```
Comment (romitgirdhar) [inline Peoplesoft/variables.tf]: Are you using this variable? This variable is for those times when your VNET and GW may be in a different Resource group as the app RG

File: Peoplesoft/main.tf [hunk 4]
```diff
@@ -184,14 +216,22 @@ resource "azurerm_resource_group" "rg" {
 ############################################################################################
 # Create the virtual network 
 
-resource "azurerm_virtual_network" "vnet" {
+data "azurerm_virtual_network" "primary_vnet" {
+    name = "${var.vnet_name}"
+    resource_group_name = "${azurerm_resource_group.rg.name}"
```
[the remark this answers is not in the document]
Comment (techbunny) [reply Peoplesoft/main.tf]: Fixed.

File: Peoplesoft/main.tf [hunk 5]
```diff
@@ -184,14 +216,22 @@ resource "azurerm_resource_group" "rg" {
...
+    count = "${var.vnet_cidr == "0" ? 1 : 0}"
+}
+
+resource "azurerm_virtual_network" "primary_vnet" {
   name                = "${var.vnet_name}"
   resource_group_name = "${azurerm_resource_group.rg.name}"
```
[the remark this answers is not in the document]
Comment (techbunny) [reply Peoplesoft/main.tf]: Fixed.

File: Peoplesoft/variables.tf (hunk 3 above)
Comment (techbunny) [reply Peoplesoft/variables.tf]: Fixed.

**Pull request merged**